### PR TITLE
docs: update LOTUS_RELEASE_FLOW.MD document

### PR DESCRIPTION
**Pull request by rjan90**
## Related Issues
#12072

## Proposed Changes
Update the LOTUS_RELEASE_FLOW document to account for the new release flow.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [x] Commits have a clear commit message.
- [x] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [x] Update CHANGELOG.md or signal that this change does not need it.
  - If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
  - If the change does not require a CHANGELOG.md entry, do one of the following:
    - Add `[skip changelog]` to the PR title
    - Add the label `skip/changelog` to the PR
- [x] New features have usage guidelines and / or documentation updates in
  - [x] [Lotus Documentation](https://lotus.filecoin.io)
  - [x] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: LOTUS_RELEASE_FLOW.md [hunk 1]
```diff
@@ -24,119 +24,84 @@
...
+Golang tightly couples source code with versioning (major versions beyond v1 leak into import paths). This poses logistical difficulties to using major versions here. Using major versions for every network upgrade would disrupt every downstream library/application that consumes the native Lotus API, even if it brought zero expectation of breakage for the Golang APIs they depend on.
 
 ## Related Items
 
-1. [Release Issue template](https://github.com/filecoin-project/lotus/blob/master/documentation/misc/RELEASE_ISSUE_TEMPLATE.md)
-2. [Lotus Release Flow Discussion](https://github.com/filecoin-project/lotus/discussions/7053): Leave a comment if you have any questions or feedbacks with regard to the lotus release flow.
+1. Release Issue template
```
Comment (BigLep) [inline LOTUS_RELEASE_FLOW.md:106]:
```suggestion
1. [Release Issue template](https://github.com/filecoin-project/lotus/blob/master/documentation/misc/RELEASE_ISSUE_TEMPLATE.md)
2. [Network Upgrade process and tracking documents for nv23 onwards](https://drive.google.com/drive/folders/1LEfIQKsp0un3RxHdSbkBG7ON3H7HRuA7)
3. [Lotus release issues](https://github.com/filecoin-project/lotus/issues?q=is%3Aissue+release+in%3Atitle)
```

I think we should leave the original hyperlink for issue template.
I think it makes to point to network upgrade docs as that gives visibility into RC timing.
Release issues also give good insight here.

File: LOTUS_RELEASE_FLOW.md [hunk 2]
```diff
@@ -24,119 +24,84 @@
...
+2. Lotus Release Flow Discussion: Leave a comment if you have any questions or feedback regarding the lotus release flow.
```
Comment (BigLep) [inline LOTUS_RELEASE_FLOW.md:107]:
```suggestion
4. [202405 discussion](https://github.com/filecoin-project/lotus/discussions/12020) that triggered updating this document.
5. [Original 202108 Lotus release flow discussion](https://github.com/filecoin-project/lotus/discussions/7053).
```

**Review by BigLep — COMMENTED** [review]
Good stuff here!  Thanks for moving this forward.  A few things:
1. I have some more sizable additions in https://github.com/filecoin-project/lotus/pull/12364 .  (A lot of it is content I saw you write elsewhere.)
2. I didn't see any discussion about "lotus node" vs. "lotus miner" here.  
3. I didn't see the TOC update.  (I assume we use a standard vscode plugin to update this automatically?)

File: LOTUS_RELEASE_FLOW.md [hunk 3]
```diff
@@ -24,119 +24,84 @@
...
-Patch releases in odd minor releases (1.3.0, 1.5.0, etc.) like 1.3.1, 1.5.2 and etc are corresponding to another **feature releases** with new development and bugfixes. These releases are not mandatory, but still **highly** recommended, **as they may contain critical security fixes**.
+We aim to ship a new release of the Lotus software approximately every 4 weeks. However, releases that include network upgrades may have longer development and testing periods.
 
----
+## Release Process
 
-### Performing a Release
+1. Releases are branched from the master branch, regardless of whether they include a network upgrade or not.
+2. All PRs should target the master branch, and if they need to be backported to a release candidate, they should be marked with a `backport` label.
+3. The `releases` branch are no longer used.
```
[the remark this answers is not in the document]
Comment (rjan90) [reply LOTUS_RELEASE_FLOW.md:83]: AI: Provide a FAQ on how to get the latest release for node|miner

**Comment by BigLep** [comment]
@rjan90 : FYI that I applied all my suggestions so it would be easier for me to make additional changes.  The "conversations" are still open so you can review the changes and adjust as desired.

**Comment by BigLep** [comment]
@rjan90 : I pushed some additional commits covering:
1. Discussion about "lotus node" vs. "lotus miner".
2. TOC handling
3. More deprecating `releases` discussion.
4. Other edits that saw

I did this here rather than a separate PR so I could link related deprecated `releases` work to just one PR.  Feel free to undo / change anything I pushed.

Other things coming to mind 2024-08-09:
1. Maybe move this file into same place that the RELEASE_ISSUE_TEMPLATE file is rather than in top-level lotus?  (Maybe good to do afterwards so the diff stays clean?)
2. Maybe link to this from CONTRIBUTING.md and/or README?

**Comment by BigLep** [comment]
> double spaces after a period

I was intentionally adding those because that was how I was taught to type 30 years ago.  Searching "english double space after period" is hilarious though.  I particularly liked this headline: "[Nothing Says Over 40 Like Two Spaces after a Period!](https://www.cultofpedagogy.com/two-spaces-after-period/)". I will have to kill this habit (and even typing this post I see myself doing it!)

> grammatical fixes

I'll figure out a grammar checker within the vscode markdown editor so I can more of these myself.

Thanks for the help!

**Review by BigLep — APPROVED** [review]
I personally am good with the content we have and am adding an Approve in case that helps with merging.  No worries though if there is additional changes you want to make.

**Pull request merged**